sway-ir/src/instruction.rs

```diff
@@ -99,6 +109,7 @@ impl Instruction {
         match self {
             Instruction::AsmBlock(asm_block, _) => asm_block.get_type(context),
             Instruction::Call(function, _) => Some(context.functions[function.0].return_type),
+            Instruction::ContractCall { .. } => None, // TODO fixed this
```

Comment:
@otrho thoughts on this? I don't have the type of this here so I probably need to carry it over in the AST. For function call, you've done this by looking at the body of the function definition. For contract calls, we can't really do that.

Reply:
From the Sway side the call returns a typed value though, which can be determined by the ABI I guess.  So we might need to either bundle up the return type into the instruction since we have the selector there, it can be fetched from the ABI.

Reply:
Note that not all contract calls will have access to the ABI at compile time. E.g. a smart contract wallet that simply calls a provided contract after authentication is not going to know the ABI of all possible contract it will call when it's compiled. _ABI casting_ implies knowing the ABI, but a use of the `CALL` instruction does not.